feature-libs/cart/main/components/events/add-to-cart-dialog-event.listener.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import {
+  ActiveCartFacade,
+  CartUiEventAddToCart,
+} from '@spartacus/cart/main/root';
+import { EventService } from '@spartacus/core';
+import { ModalRef, ModalService } from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+import { AddedToCartDialogComponent } from '../cart/add-to-cart/added-to-cart-dialog/added-to-cart-dialog.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AddToCartDialogEventListener implements OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected activeCartFacade: ActiveCartFacade,
+    protected eventService: EventService,
+    protected modalService: ModalService
+  ) {
+    this.onAddToCart();
+  }
+
+  protected onAddToCart() {
+    this.subscription.add(
+      this.eventService.get(CartUiEventAddToCart).subscribe((event) => {
+        this.openModal(event);
+      })
+    );
+  }
+  protected openModal(event: CartUiEventAddToCart): void {
+    let modalRef: ModalRef;
+
+    let modalInstance: any;
+    modalRef = this.modalService.open(AddedToCartDialogComponent, {
```

Comment:
`const modalRef = ...`, and remove `let modalRef: ModalRef;`

Reply:
fixed